taxi/views.py

```diff
@@ -84,6 +137,30 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
 class DriverListView(LoginRequiredMixin, generic.ListView):
     model = Driver
     paginate_by = 5
+    context_object_name = "driver_list"
+
+    def get_context_data(self, *, object_list=None, **kwargs):
+        contex = super(DriverListView, self).get_context_data(**kwargs)
+
+        username = self.request.GET.get("username", "")
+
+        contex["search_form"] = DriverSearchForm(initial={
+            "username": username
+        })
+
+        return contex
+
+    def get_queryset(self):
+        queryset = Driver.objects.all()
+
+        form = DriverSearchForm(self.request.GET)
+
+        if form.is_valid():
+            return queryset.filter(
+                username__icontains=form.cleaned_data["username"]
+            )
+
+        return queryset
```

Comment:
Code duplication for `get_context_data` and `get_queryset` methods here, in `CarListView`, and `ManufacturerListView`. Think about unifying the logic and adding it in a form of a Mixin to these views.